api/handlers.go

```diff
@@ -639,7 +639,7 @@ func (si *ServerImplementation) fetchAccounts(ctx context.Context, options idb.A
 
 		// Compute for a given round if requested.
 		var account generated.Account
-		if atRound != nil {
+		if (atRound != nil) && (*atRound < row.Account.Round) {
```

Comment:
requesting a round in the future could return an error if `atRound > round` at line 623

Reply:
Yes, let's do it in a different pull request.